pandas/core/series.py

```diff
@@ -2185,18 +2185,33 @@ def _binop(self, other, func, level=None, fill_value=None):
 
         this_vals, other_vals = ops.fill_binop(this.values, other.values,
                                                fill_value)
-
-        with np.errstate(all='ignore'):
-            result = func(this_vals, other_vals)
         name = ops.get_op_result_name(self, other)
+
+        if (is_extension_array_dtype(this_vals) or
```

Comment:
Stepping back a bit here, why does ExtensionArray get a fallback to `[func(a, b) for a, b in zip(...)]`, when ndarray doesn't? This confuses the expected signature for `func`: does it that two arrays `a` and `b`, or two scalars?

Reply:
Indeed. 
And I also don't really understand how this is related to `combine`, as `_binop` does not seem to be used there. I think `_binop` is for implementing the flex arithmetic ops like `.add()` etc. But for that I think we should do this together with the actual ops change, and probably end up with the same conclusion (no explicit element-wise fallback but rely on the ExtensionArray implementation to do this)

Reply:
agree here. I am puzzled why this change at all. If the extension array supports ops this would work as is, if not it would raise a TypeError. So not sure what you are trying to do here.

Reply:
My mistake.  When I separated out this PR from #20889, I should have put back the old `_binop`.  Fixed in my next commit.